### ISSUE TITLE
Display "edition" from manifest on crate-page

### ISSUE THREAD

**Issue by lukaslueg** [rust-lang/crates.io#1541]
[This](https://crates.io/crates/actix-web-async-await) is an example of a crate that has an explicit `edition = "2018"` in it's manifest. It would be nice if the crate-page would show a marker/banner/hint that this crate expects a 2018-environment

**Comment by Mark-Simulacrum** [comment]
This just requires a recent enough compiler so I don't think this is something we should do this as it's really no different than "this crate requires 1.31" which is a separate discussion (there have been multiple RFCs on that)

**Comment by carols10cents** [comment]
@lukaslueg and @Eh2406, could you elaborate a bit on why this would be nice/useful and different from the compiler version the crate requires? 

Given a new enough compiler, you can depend on crates that use either edition without needing to care which edition the dependency uses, barring some potential keywords used as functions in the public API that would need the raw identifier syntax.

Or do you mean as a potential contributor to the crate, rather than a potential user of it?

In other words, if this information was displayed on crates.io, what would you do differently based on it?

**Comment by Eh2406** [comment]
I had not fully thought this thure when I emoted, having read the responses I agree that it is just a weaker form of MSRV.

**Comment by lukaslueg** [comment]
Well, this is actually a weak argument in the first place: While there is an ongoing discussion on how to communicate rustc version requirements, the `edition`-key in the manifest is as clear as it could: Either it's  there (because you have to have it in order to use `2018`) or it's not there (which gets us "needs rust 1.28ish"). Since it's also readily available from the manifest, it's very easy to implement without any kind of ambiguity. Besides - and the arguments get even weaker here - the whole `edition` concept is probably less known to the wider audience than one may think. A hint on `crates.io` could help communicate "hey, this is a thing!"

**Comment by meven** [comment]
I think it would make sense, for the users to know about editions as @lukaslueg mentioned and to advertise the lattest edition (in case someone missed the information) and slightly encourage maintainers to migrate to a more recent edition because it would grant their crate the edtion 2018 badge.
But also because a byproduct of this could be to have statistics about the number of "edtion 2018" crates and crates "edtion 2015".

**Comment by kornelski** [comment]
FWIW I've implemented that on crates.rs: https://crates.rs/crates/actix-web-async-await

<img width="176" alt="screenshot 2019-01-09 at 15 46 45" src="https://user-images.githubusercontent.com/72159/50910483-ca300180-1425-11e9-93c0-76c760599ecf.png">

**Comment by Turbo87** [comment]
in any case we will want the information in the database at least. how exactly we display it or what we derive from it we can still discuss once we actually have the information :)

**Comment by epage** [comment]
I'm curious, what is helped by having it in the database?

**Comment by Turbo87** [comment]
at the very least it will make it easier to run edition data analysis on the database dump, but as you wrote above, inferring an MSRV from the edition field is also a possibility

**Comment by epage** [comment]
So last this was discussed, the next step was to put it in the database "for reasons" without an end plan for this. Now a PR is merged to show it without any reasoning given in the PR. Yes, its at least in a tooltip so its not taking up space but being tied tg the MSRV tooltip and saying the package requires an Edition at best does nothing (they already have an MSRV) and at worse implies some additionally requirement beyond MSRV.

**Comment by Turbo87** [comment]
some people find it informational and I don't see why we should completely hide the information. if you are not one of those people then that's fine, but I don't see what's wrong with showing this information in the tooltip, since, as you requested above, it's not even using up the precious real estate on the screen.
